### PR TITLE
Add go backend

### PR DESCRIPTION
**Pull request by sobel-stripe**
Adds a go backend as an `Echo` app

### CODE REVIEW

File: server/go/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+## Getting Started
+
+Copy the example environment variables file `.env.example` from the root of the repo into your own environment file called `.env`:
+
+```
+cp .env.example .env
+```
+
+
+Run the application from this directory
```
Comment (thorsten-stripe) [inline server/go/README.md]: can you include the `cd server/go/` command, to make sure folks understand the context of "this directory"?

File: server/go/app.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+			products, err = inventory.ListProducts()
+			if err != nil {
+				return err
+			}
+		}
+
+		return c.JSON(http.StatusOK, listing{products})
+	})
+
+	e.GET("/product/:product_id/skus", func(c echo.Context) error {
```
Comment (thorsten-stripe) [inline server/go/app.go]: I just realised that this should be plural which is an issue in the frontend. No action needed from you, just a reminder for me to update throughout.

File: server/go/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Stripe Payments Demo - Go Server
+
+This demo uses a simple [Echo](https://echo.labstack.com/) application as the server.
+
+## Payments Integration
+
+- [`app.go`](app.go) contains the routes that interface with Stripe to create
```
Comment (thorsten-stripe) [inline server/go/README.md]: Is this sentence incomplete?

File: server/go/payments/payments.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+		return nil, fmt.Errorf("payments: error updating payment intent: %v", err)
+	}
+
+	return pi, nil
+}
+
+func paymentMethodTypes() []*string {
+	types := config.PaymentMethods()
+
+	out := make([]*string, len(types))
```
Comment (thorsten-stripe) [inline server/go/payments/payments.go]: @sobel-stripe I'm running into a weird case where the payment methods on the PI are all of type wechat: pi_1EQBSIH9h3VSbUJPCH90Ff3N. Any idea why?

Comment (sobel-stripe) [reply server/go/payments/payments.go]: ah yep — ran into https://developmentality.wordpress.com/2014/02/25/go-gotcha-0-why-taking-the-address-of-an-iterated-variable-is-wrong/ while converting the PaymentMethods `[]string` to `[]*string`. good catch

**Comment by thorsten-stripe** [comment]
Great stuff, thanks @sobel-stripe

**Pull request merged**